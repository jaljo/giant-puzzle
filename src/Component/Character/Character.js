import React from 'react'
import './Character.scss'

// Character :: Props -> React.Component
export default ({
  asset,
  direction,
  isBurrowed,
}) =>
  <div
    data-is="character"
    className={`${asset} ${direction} ${isBurrowed ? 'is-burrowed' : ''}`}
  >
  </div>
