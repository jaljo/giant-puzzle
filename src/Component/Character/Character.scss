div[data-is="character"] {
  width: 128px;
  height: 128px;

  &.chick {
    background-image: url('/images/chicky.png');
  }

  &.fox {
    background-image: url('/images/foxy.png');
  }

  &.up {
    background-position-x: 0px;
  }

  &.down {
    background-position-x: -384px;
  }

  &.left {
    background-position-x: -256px;
  }

  &.right {
    background-position-x: -128px;
  }
}
