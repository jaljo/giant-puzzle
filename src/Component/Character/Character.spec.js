import React from 'react'
import renderer from 'react-test-renderer'
import Character from './Character'

describe('Component :: Character', () => {
  it('renders a character component', () => {
    const state = {
      direction: 'up',
      asset: 'fox',
      isBurrowed: false,
    }

    expect(
      renderer.create(<Character {...state} />)
    ).toMatchSnapshot()
  })

  it('renders a burrowed character component', () => {
    const state = {
      direction: 'down',
      asset: 'chick',
      isBurrowed: true,
    }

    expect(
      renderer.create(<Character {...state} />)
    ).toMatchSnapshot()
  })
})
