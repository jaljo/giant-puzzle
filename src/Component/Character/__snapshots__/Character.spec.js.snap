// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component :: Character renders a burrowed character component 1`] = `
<div
  className="chick down is-burrowed"
  data-is="character"
/>
`;

exports[`Component :: Character renders a character component 1`] = `
<div
  className="fox up "
  data-is="character"
/>
`;
